pkgs/by-name/yt/ytmdesktop/package.nix

```diff
@@ -0,0 +1,97 @@
+{
+  lib,
+  asar,
+  binutils,
+  commandLineArgs ? "",
+  electron_30,
+  fetchurl,
+  makeDesktopItem,
+  makeWrapper,
+  nix-update-script,
+  stdenvNoCC,
+  zstd,
+}:
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "ytmdesktop";
+  version = "2.0.5";
+
+  meta = {
+    changelog = "${finalAttrs.meta.downloadPage}/tag/v${finalAttrs.version}";
+    description = "A Desktop App for YouTube Music";
+    downloadPage = "https://github.com/ytmdesktop/ytmdesktop/releases";
+    homepage = "https://ytmdesktop.app/";
+    license = lib.licenses.gpl3Only;
+    mainProgram = finalAttrs.pname;
+    maintainers = [ lib.maintainers.cjshearer ];
+    inherit (electron_30.meta) platforms;
+    # While the files we extract from the .deb are cross-platform (javascript), the installation
+    # process for darwin is different, and I don't have a test device. PRs are welcome if you can
+    # add the correct installation steps. I would suggest looking at the following:
+    # https://www.electronjs.org/docs/latest/tutorial/application-distribution#manual-packaging
+    # https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/networking/instant-messengers/jitsi-meet-electron/default.nix
+    badPlatforms = lib.platforms.darwin;
+  };
+
+  passthru.updateScript = nix-update-script { };
+
+  desktopItem = makeDesktopItem {
+    desktopName = "Youtube Music Desktop App";
+    exec = finalAttrs.pname;
+    icon = finalAttrs.pname;
+    name = finalAttrs.pname;
+    genericName = finalAttrs.meta.description;
+    mimeTypes = [ "x-scheme-handler/ytmd" ];
+    categories = [
+      "AudioVideo"
+      "Audio"
+    ];
+    startupNotify = true;
+  };
+
+  nativeBuildInputs = [
+    asar
+    binutils
+    electron_30
```

Comment:
[Disclaimer, I know very little about Linux Application building / NixOS as a whole so stop me if I'm being 100% an idiot]
 
This version of the application is on v29 of Electron and not v30?
We've tested and delayed our release of using v30 as there has been a number of issues with YTMDesktop on v30 due to the usage of bits which recently have been marked for deprecation. Otherwise there were a few things which entirely weren't working.

Closing the application on Windows at least prompts you to with the Navigation away.
You can see the commit reverting back to v29 here: https://github.com/ytmdesktop/ytmdesktop/commit/6da08757645dd894fab585ae3f4aa73cc9b60ec2

Reply:
Oops, yeah my bad. I'll drop it back to 29.

Reply:
Wait a second. I remember why I did that. Electron 29 is missing in nixpkgs unstable. I'm not actually sure why. Should I drop it back to 28 or leave it at 30?

Reply:
It should be fine either way. Issues with 30 only occurred when actually building against 30. Since we're using Debian binaries compiled with 29 we should be good since it ran fine for me